### PR TITLE
Fix unencoded TPCH Benchmarks

### LINKED ISSUE #949: Unencoded TPC-H fails
```
./hyriseBenchmarkTPCH -e Unencoded
[PERF] Hyrise is running as a debug build. at src/lib/utils/performance_warning.cpp:13
	Performance can be affected. This warning is only shown once.

libc++abi.dylib: terminating with uncaught exception of type std::logic_error: src/lib/statistics/chunk_statistics/chunk_statistics.cpp:10 The statistics should not contain any empty shared_ptrs.
Abort trap: 6
```

### LINKED ISSUE #957: Chunk::is_mutable()
...is currently defined as 

```c++

bool Chunk::is_mutable() const {
  return std::all_of(_columns.begin(), _columns.end(),
                     [](const auto& column) { return std::dynamic_pointer_cast<BaseValueColumn>(column) != nullptr; });
}
```

however, @Bouncner pointed out in https://github.com/hyrise/hyrise/pull/956#issuecomment-402512336, that chunks might still contain ValueColumns and still be immutable - e.g. (i.e.?) if they have column statistics.

Do we want `is_mutable()` to return `(bool)_statistics`? and rely on this implicit connection?

### PR DESCRIPTION
**Pull request by lawben**
Fixes #949, fixes #957.

### CODE REVIEW

**Comment by lawben** [comment]
Ah, okay. I'll have a look into that. This was just supposed to be a quick fix for the error you were getting.

**Comment by mrzzzrm** [comment]
@Bouncner we prune based on the ChunkColumnStatistics. That's why we can't have them on non-frozen data

**Comment by Bouncner** [comment]
But we do not encode mutable chunks. In my (very limited) understanding, encoding a chunk means that this chunk will be immutable.
And having an immutable chunk being unencoded is totally fine.

**Comment by mrzzzrm** [comment]
then I guess only the encoding should be optional, not the line below it that gathers the statistics.

File: src/lib/statistics/chunk_statistics/chunk_column_statistics.cpp [hunk 1]
```diff
@@ -53,22 +53,18 @@ std::shared_ptr<ChunkColumnStatistics> ChunkColumnStatistics::build_statistics(D
...
-            values.insert(value.value());
-          }
-        });
-        pmr_vector<DataTypeT> dictionary{values.cbegin(), values.cend()};
-        std::sort(dictionary.begin(), dictionary.end());
-        statistics = build_statistics_from_dictionary(dictionary);
-      } else {
-       Fail("ChunkColumnStatistics should only be built for encoded columns.");
-      }
+      // if we have a generic encoded column we create the dictionary ourselves
```
Comment (mrks) [inline src/lib/statistics/chunk_statistics/chunk_column_statistics.cpp]: is this comment still correct? we are not only looking at encoded columns, are we?

**Comment by lawben** [comment]
@mrks I ran this with FullCI but is it still waiting for report. i could just restart it, but maybe there is still a bug somewhere?

**Comment by mrks** [comment]
Did you merge master? If you are still running on the old Jenkinsfile, the notification won't happen.

**Comment by lawben** [comment]
For the record: same coverage problem as in #953. Down 17.35% to 68.65%

**Comment by lawben** [comment]
... and the coverage diff shown here in the checks is now dead (404), so it should probably be removed.

**Comment by mrks** [comment]
I'll sneak this in here if you don't mind.

**Pull request merged**